plugins/memblaze/memblaze-nvme.c

```diff
@@ -1840,3 +1900,529 @@ static int mb_get_latency_feature(int argc, char **argv, struct command *cmd,
+		[12] = {"900us", "1ms"  },
+		[13] = {"1ms",   "5ms"  },
+		[14] = {"5ms",   "10ms" },
+		[15] = {"10ms",  "20ms" },
+		[16] = {"20ms",  "50ms" },
+		[17] = {"50ms",  "100ms"},
+		[18] = {"100ms", "200ms"},
+		[19] = {"200ms", "300ms"},
+		[20] = {"300ms", "400ms"},
+		[21] = {"400ms", "500ms"},
+		[22] = {"500ms", "600ms"},
+		[23] = {"600ms", "700ms"},
+		[24] = {"700ms", "800ms"},
+		[25] = {"800ms", "900ms"},
+		[26] = {"900ms", "1s"   },
+		[27] = {"1s",    "2s"   },
+		[28] = {"2s",    "3s"   },
+		[29] = {"3s",    "4s"   },
+		[30] = {"4s",    "5s"   },
+		[31] = {"5s",    "8s"   },
+		[32] = {"8s",    "INF"  },
+	};
+
+	printf("Bucket 1-32 IO Read Command Data\n");
+	printf("-------------------------------------------\n");
+	printf("%-12s%-12s%-12s%-12s\n", "Bucket", "Start(>=)", "End(<)", "Value");
+	int bucket_count = sizeof(log->v2_0.bucket_read_data) / sizeof(uint32_t);
+
+	for (int i = 0; i < bucket_count; i++) {
+		printf("%-12u%-12s%-12s%-12u\n", i + 1, buckets[i + 1].start_threshold,
+			   buckets[i + 1].end_threshold, log->v2_0.bucket_read_data[i]);
+	}
+	printf("\n");
+
+	printf("Bucket 1-32 IO Write Command Data\n");
+	printf("-------------------------------------------\n");
+	printf("%-12s%-12s%-12s%-12s\n", "Bucket", "Start(>=)", "End(<)", "Value");
+	bucket_count = sizeof(log->v2_0.bucket_write_data) / sizeof(uint32_t);
+
+	for (int i = 0; i < bucket_count; i++) {
+		printf("%-12u%-12s%-12s%-12u\n", i + 1, buckets[i + 1].start_threshold,
+			   buckets[i + 1].end_threshold, log->v2_0.bucket_write_data[i]);
+	}
+	printf("\n");
+
+	printf("Bucket 1-32 IO Trim Command Data\n");
+	printf("-------------------------------------------\n");
+	printf("%-12s%-12s%-12s%-12s\n", "Bucket", "Start(>=)", "End(<)", "Value");
+	bucket_count = sizeof(log->v2_0.bucket_trim_data) / sizeof(uint32_t);
+
+	for (int i = 0; i < bucket_count; i++) {
+		printf("%-12u%-12s%-12s%-12u\n", i + 1, buckets[i + 1].start_threshold,
+			   buckets[i + 1].end_threshold, log->v2_0.bucket_trim_data[i]);
+	}
+	printf("\n");
+}
+
+static void latency_stats_print(struct latency_stats *log, const char *devname)
+{
+	uint32_t minor_version = *(uint32_t *)&log->raw[0];
+	uint32_t major_version = *(uint32_t *)&log->raw[4];
+
+	printf("Major Version: %u, Minor Version: %u\n", major_version, minor_version);
+	printf("\n");
+	printf("Latency Statistics Log for NVMe device: %s\n", devname);
+	printf("\n");
+
+	switch (major_version) {
+	case 2:
+		switch (minor_version) {
+		case 0:
+			latency_stats_v2_0_print(log, sizeof(struct latency_stats));
+			break;
+		default:
+			fprintf(stderr, "Major Version %u, Minor Version %u: Not supported yet\n",
+					major_version, minor_version);
+			break;
+		}
+		break;
+
+	default:
+		fprintf(stderr, "Major Version %u: Not supported yet\n", major_version);
+		break;
+	}
+}
+
+static int mb_get_latency_stats(int argc, char **argv,
+								struct command *cmd,
+								struct plugin *plugin)
+{
+	// Get the configuration
```

Comment:
Please avoid non meaningful comments. It's obvious from the code what is happening. This clutters just the code. Some for the rest of the comments in this function.